### PR TITLE
IGNITE-14298 DEVNOTES should contain clear instructions on how to run check-style

### PR DESCRIPTION
**Pull request by vveider**

### CODE REVIEW

File: DEVNOTES.md [hunk 1]
```diff
@@ -1,99 +1,119 @@
...
+[Checkstyle suppressions](check-rules/checkstyle-suppressions.xml).
+```
+mvn clean checkstyle:checkstyle-aggregate
+```
+Code style check result is generated at `target/site/checkstyle-aggregate.html`
+
+### License headers
+Project files license headers match with required template is checked with [Apache Rat Maven Plugin](https://creadur.apache.org/rat/apache-rat-plugin/).
+```
+mvn clean apache-rat:check -pl :apache-ignite
```
Comment (agura) [inline DEVNOTES.md:38]: Execute `mvn clean apache-rat:check -pl :apache-ignite` from the project root.

Result FAILED:

```
[ERROR] Failed to execute goal org.apache.rat:apache-rat-plugin:0.13:check (default-cli) on project apache-ignite: Too many files with unapproved license: 12 See RAT report in: C:\git\ignite-3\target\rat.txt -> [Help 1]
```

`rat.txt` snippet:

```
Files with unapproved licenses:

  modules/cli/ignite-cli.iml
  modules/cli-common/ignite-cli-common.iml
  modules/configuration/ignite-configuration.iml
  modules/configuration-annotation-processor/ignite-configuration-annotation-processor.iml
  modules/metastorage-client/metastorage-client.iml
  modules/metastorage-common/metastorage-common.iml
  modules/network/ignite-network.iml
  modules/raft/ignite-jraft.iml
  modules/raft/raft.iml
  modules/raft-client/ignite-raft-client.iml
  modules/rest/ignite-rest.iml
  modules/runner/ignite-runner.iml
```

So `*.iml` files should be excluded from target files.

File: DEVNOTES.md [hunk 2]
```diff
@@ -1,99 +1,119 @@
...
+```
+License headers check result is generated at `target/rat.txt`
+
+### PMD
+Static code analyzer is run with [Apache Maven PMD Plugin](https://maven.apache.org/plugins/maven-pmd-plugin/). Precompilation is required 'cause PMD shoud be run on compiled code.
+
+[PMD rules](check-rules/pmd-rules.xml).
+```
+mvn clean compile pmd:check
```
Comment (agura) [inline DEVNOTES.md:47]: Execute `mvn clean compile pmd:check` from the project root.

Result FAILED:

```
 Failed to execute goal org.apache.maven.plugins:maven-pmd-plugin:3.14.0:check (default-cli) on project apache-ignite: You have 5 PMD violations. For more details see: C:\git\gridgain\ignite-3\target\pmd.xml -> [Help 1]
```

`pmd.xml` snippet:

```

<?xml version="1.0" encoding="UTF-8"?>
<pmd xmlns="http://pmd.sourceforge.net/report/2.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://pmd.sourceforge.net/report/2.0.0 http://pmd.sourceforge.net/report_2_0_0.xsd" version="6.29.0" timestamp="2021-03-10T21:28:42.978">
<file name="C:\git\gridgain\ignite-3\modules\rest\target\generated-sources\annotations\org\apache\ignite\rest\configuration\impl\RestNode.java">
<violation beginline="16" endline="16" begincolumn="43" endcolumn="47" rule="UnusedPrivateField" ruleset="Best Practices" package="org.apache.ignite.rest.configuration.impl" class="RestNode" variable="_spec" externalInfoUrl="https://pmd.github.io/pmd-6.29.0/pmd_rules_java_bestpractices.html#unusedprivatefield" priority="3">
Avoid unused private fields such as '_spec'.
</violation>
</file>
<file name="C:\git\gridgain\ignite-3\modules\runner\target\generated-sources\annotations\org\apache\ignite\configuration\extended\impl\AutoAdjustNode.java">
<violation beginline="17" endline="17" begincolumn="49" endcolumn="53" rule="UnusedPrivateField" ruleset="Best Practices" package="org.apache.ignite.configuration.extended.impl" class="AutoAdjustNode" variable="_spec" externalInfoUrl="https://pmd.github.io/pmd-6.29.0/pmd_rules_java_bestpractices.html#unusedprivatefield" priority="3">
Avoid unused private fields such as '_spec'.
</violation>
</file>
<file name="C:\git\gridgain\ignite-3\modules\runner\target\generated-sources\annotations\org\apache\ignite\configuration\extended\impl\BaselineNode.java">
<violation beginline="18" endline="18" begincolumn="47" endcolumn="51" rule="UnusedPrivateField" ruleset="Best Practices" package="org.apache.ignite.configuration.extended.impl" class="BaselineNode" variable="_spec" externalInfoUrl="https://pmd.github.io/pmd-6.29.0/pmd_rules_java_bestpractices.html#unusedprivatefield" priority="3">
Avoid unused private fields such as '_spec'.
</violation>
</file>
<file name="C:\git\gridgain\ignite-3\modules\runner\target\generated-sources\annotations\org\apache\ignite\configuration\extended\impl\DataStorageNode.java">
<violation beginline="16" endline="16" begincolumn="50" endcolumn="54" rule="UnusedPrivateField" ruleset="Best Practices" package="org.apache.ignite.configuration.extended.impl" class="DataStorageNode" variable="_spec" externalInfoUrl="https://pmd.github.io/pmd-6.29.0/pmd_rules_java_bestpractices.html#unusedprivatefield" priority="3">
Avoid unused private fields such as '_spec'.
</violation>
</file>
<file name="C:\git\gridgain\ignite-3\modules\runner\target\generated-sources\annotations\org\apache\ignite\configuration\extended\impl\LocalNode.java">
<violation beginline="20" endline="20" begincolumn="44" endcolumn="48" rule="UnusedPrivateField" ruleset="Best Practices" package="org.apache.ignite.configuration.extended.impl" class="LocalNode" variable="_spec" externalInfoUrl="https://pmd.github.io/pmd-6.29.0/pmd_rules_java_bestpractices.html#unusedprivatefield" priority="3">
Avoid unused private fields such as '_spec'.
</violation>
</file>
</pmd>
```

But on TC there are no any violations.

**Review by agura — CHANGES_REQUESTED** [review]
It seems that it still does not work correctly.

File: DEVNOTES.md (hunk 1 above)
Comment (vveider) [reply DEVNOTES.md:38]: Added  exclusion, please check.

File: DEVNOTES.md (hunk 2 above)
Comment (vveider) [reply DEVNOTES.md:47]: Merge fixes from main, now all checks are good.

**Pull request closed without merging**